### PR TITLE
Greylisting

### PR DESCRIPTION
**Pull request by mundschenk-at**
A first pass at greylisting support:
* Rmilter/Rspamd
* Postgrey

### CODE REVIEW

File: modoboa_stats/graphics.py [hunk 1]
```diff
@@ -158,8 +164,8 @@ class GraphicSet(object):
     title = None
     _graphics = []
 
-    def __init__(self):
-        self.__ginstances = []
+    def __init__(self, instances=[]):
```
Comment (tonioo) [inline modoboa_stats/graphics.py]: This is not safe. Prefer something like this:
```python
def __init__(self, instances=None):
    if instances is None:
        instances = []
```

File: modoboa_stats/handlers.py [hunk 2]
```diff
@@ -25,5 +26,6 @@ def menu(sender, location, user, **kwargs):
 @receiver(signals.get_graph_sets)
 def get_default_graphic_sets(sender, **kwargs):
     """Return graphic set."""
-    gset = graphics.MailTraffic()
+    gset = graphics.MailTraffic(param_tools.get_global_parameter(
+            "greylist", app="modoboa_stats"))
```
Comment (tonioo) [inline modoboa_stats/handlers.py]: You don't have to specify the app in this case, it will be guessed automatically.

**Comment by mundschenk-at** [comment]
@tonioo Fixed.

**Comment by tonioo** [comment]
@mundschenk-at Thank you. Can I merge the issue?

**Comment by mundschenk-at** [comment]
Technically? I think so. Postgrey support is based on example log messages on the web and has not been tested on a live system, though.

**Comment by tonioo** [comment]
Ok. I don't use postgrey anymore but I could add support for postscreen. What do you think?

**Comment by mundschenk-at** [comment]
That would be great!

**Comment by tonioo** [comment]
@mundschenk-at I tried something here: https://github.com/modoboa/modoboa-stats/pull/25.

I found two issues with your PR:
- Once created, it is not possible to add a new DS to a RRD file (I forgot that when I reviewed). See https://www.mail-archive.com/rrd-users@lists.oetiker.ch/msg20161.html. Looks like the 1.5 branch allows the creation of a new file based on an existing one but a lot of Linux distributions still provide the 1.4 branch...
- Postgrey reject message can be changed to a custom value so you can't use the default message as a lookup pattern.

**Comment by mundschenk-at** [comment]
Regarding customized Postgrey messages, I'd say "if you change the default, you're out of luck". I mean we could add a config section to customize regexes, but this would be a "we won't help you if you anything goes wrong with this" kinda thing.

As for the RRD files, I did some research on that beforehand, but 1.5 has been out for two years now, so I felt it somewhat safe to require that for seemlessly upgrading the RRD files. Should we add a warning if that fails (the user would have to delete the files to have them recreated with the new DS)?

**Comment by tonioo** [comment]
Could it be possible to do something similar to what I added for postgrey?

About RRDtool, you're saying that the code you added works with a version > 1.5 ?

**Comment by mundschenk-at** [comment]
RRDtool: I've RRDtool 1.6.0 installed and adding the DS via the python bindings seems to work fine, yes.

**Comment by tonioo** [comment]
@mundschenk-at I think we should indeed add a warning somewhere if RRDtool has not the required version. Any idea?

**Comment by mundschenk-at** [comment]
@tonioo Is there a way to add a help string to the parameters form?

**Comment by tonioo** [comment]
Are you talking about that:
https://github.com/modoboa/modoboa-stats/blob/master/modoboa_stats/forms.py#L20

**Comment by mundschenk-at** [comment]
Oh, I forgot about that! Yes, but I think the current implementation is probably not prominent enough. Or is there a way to check rrtools version when enabling the parameter and popping up a warning in that case?

**Comment by tonioo** [comment]
Maybe something similar to this:
https://github.com/modoboa/modoboa/blob/master/modoboa/admin/app_settings.py#L116

**Pull request merged**